src/collectors/windows.plugin/windows_plugin.c

```diff
@@ -123,9 +206,14 @@ void *win_plugin_main(void *ptr)
             if (unlikely(!pm->enabled))
                 continue;
 
+            pm->update_counter += 1;
+            if (unlikely(pm->update_counter != pm->update_every))
+                continue;
+
+            pm->update_counter = 0;
```

Comment:
This is an incorrect approach. Data collection intervals must align with second boundaries. For example, with a 30-second collection interval, data should only be collected at seconds 0 and 30 of each minute.

Ask @stelfrag for details.

Reply:
As we discussed, we will have a new PR to address this.